libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Maxell.cpp

```diff
@@ -95,6 +96,12 @@ void AP_BattMonitor_SMBus_Maxell::timer()
         _state.current_amps = -(float)((int16_t)data) / 1000.0f;
         _state.last_time_micros = tnow;
     }
+
+    // read temperature
+    if (read_word(BATTMONITOR_SMBUS_MAXELL_TEMP, data)) {
+        _state.has_temperature = true;
```

Comment:
I implemented this, but it appears to be a bad decision, I've attatched an image from a Solo battery, and it appears this part of the communication is really unreliable. The better approach I think will involve a timer that sets it to false if we haven't received a new temperature update for a period of time (5 seconds comes to mind as the number already used elsewhere in the drivers). Temp isn't a high priority information, but it would be nice to not be oscillating like this.
![figure_1-2](https://cloud.githubusercontent.com/assets/567688/24990440/ff96c79e-1fc8-11e7-9ae6-bc42d40a1692.png)